test/ffmpeg-qsv/transcode/transcoder.py

```diff
@@ -154,6 +181,26 @@ def check_output(self):
       "hwaccel initialisation returned error", self.output, re.MULTILINE)
     assert m is None, "Failed to use hardware decode"
 
+  # convert scaled test stream to yuv with source width and height
+  def scale_to_yuv(self, yuv, encoded, codec, mode):
```

Comment:
I'm not convinced that a dedicated function is necessary.  There are so many common pipeline pieces that are coded over and over in this class.  Much can be captured via code reuse.  Break it down to decode parts, filter parts, and output parts.